server/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -902,6 +902,10 @@ protected final GetResult getFromSearcher(Get get, Engine.Searcher searcher, boo
         }
     }
 
+    public boolean isDocumentInLiveVersionMap(BytesRef uid) {
```

Comment:
maybe it's better if we name this `isDocumentSearchable`? more generic and less tied to implementation details. And +1 to add more docs.